### PR TITLE
Introduce a separate `GitCommand` class

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: plugins/package-managers/gradle-inspector/src/funTest/kotlin/GradleFunTest.kt [hunk 1]
```diff
@@ -30,7 +30,7 @@ import io.kotest.matchers.should
 import org.ossreviewtoolkit.analyzer.create
 import org.ossreviewtoolkit.analyzer.resolveSingleProject
 import org.ossreviewtoolkit.model.toYaml
-import org.ossreviewtoolkit.plugins.versioncontrolsystems.git.Git
+import org.ossreviewtoolkit.plugins.versioncontrolsystems.git.GitCommand
```
Comment (fviernau) [inline plugins/package-managers/gradle-inspector/src/funTest/kotlin/GradleFunTest.kt:33]: Please provide a rationale _why_ the alignment is done in the direction chosen.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: plugins/commands/requirements/src/main/kotlin/RequirementsCommand.kt [hunk 2]
```diff
@@ -174,12 +174,20 @@ class RequirementsCommand : OrtCommand(
             Modifier.isAbstract(it.modifiers) || it.isAnonymousClass || it.isLocalClass
         }.sortedBy { it.simpleName }.forEach {
             runCatching {
+                fun isBundledPlugin(type: String) = it.packageName.startsWith("org.ossreviewtoolkit.plugins.$type")
```
Comment (fviernau) [inline plugins/commands/requirements/src/main/kotlin/RequirementsCommand.kt:177]: Nit: Shall the argument of `startWith` end with an (additional) `.`?

File: plugins/commands/requirements/src/main/kotlin/RequirementsCommand.kt [hunk 3]
```diff
@@ -174,12 +174,20 @@ class RequirementsCommand : OrtCommand(
...
                 var category = "Other tool"
                 val instance = when {
                     kotlinObject != null -> {
                         logger.debug { "$it is a Kotlin object." }
+
+                        when {
+                            isBundledPlugin("packagemanagers") -> category = "PackageManager"
+                            isBundledPlugin("versioncontrolsystems") -> category = "VersionControlSystem"
+                        }
+
```
Comment (fviernau) [inline plugins/commands/requirements/src/main/kotlin/RequirementsCommand.kt:190]: Have you considered extracting a `fun CommandLineTool.getCategory()` ? To speparate concerns a bit?

Comment (sschuberth) [reply plugins/commands/requirements/src/main/kotlin/RequirementsCommand.kt:191]: Done, but slightly differently.

**Review by fviernau — DISMISSED** [review]

**Pull request merged**